libraries/AP_RangeFinder/AP_RangeFinder_HC_SR04.cpp

```diff
@@ -116,35 +79,19 @@ bool AP_RangeFinder_HC_SR04::detect(AP_RangeFinder_Params &_params)
 }
 
 
-// interrupt handler for reading distance-proportional time interval
-void AP_RangeFinder_HC_SR04::irq_handler(uint8_t pin, bool pin_high, uint32_t timestamp_us)
-{
-    if (pin_high) {
-        pulse_start_us = timestamp_us;
-    } else {
-        irq_value_us = timestamp_us - pulse_start_us;
-    }
-}
-
 /*
   update distance_cm
  */
 void AP_RangeFinder_HC_SR04::update(void)
 {
     // check if pin has changed and configure interrupt handlers if required:
-    check_pins();
-
-    if (!echo_pin || ! trigger_pin || echo_pin == -1 || trigger_pin == -1) {
+    if (!check_pins()) {
```

Comment:
is this a behaviour change for pin values of -1?

Reply:
Checks have moved.